spec/controllers/events_controller_spec.rb

```diff
@@ -1,29 +1,188 @@
 require 'rails_helper'
 
 RSpec.describe EventsController, type: :controller do
-  describe 'POST events#create' do
-    let!(:admin) { FactoryBot.create(:admin) }
+  describe 'actions' do
+    let(:admin) { create(:admin) }
+    let(:user) { create(:user) }
+
+    let(:event_params) do
+      {
+        description: 'Desccription for event',
+        status: 'scheduled',
+        start_date: Time.zone.local(2019, 'jan', 1),
+        finish_date: Time.zone.local(2019, 'jan', 2),
+        comments: 'Comment about event',
+        summary: 'Summary about event',
+        user_id: user.id,
+        targets_attributes: [
+          { description: 'Test data' },
+          { description: 'Test data 2' }
+        ]
+      }
+    end
 
     before(:each) do
       login_user admin
     end
 
-    it 'send email when create event' do
-      params = { event: {
-        description: 'some description',
-        start_date: Time.current - 1.day,
-        finish_date: Time.current, user_id: admin.id
-      } }
+    context 'POST :create' do
+      it 'create with valid params' do
+        expect do
+          post :create, params: { event: event_params }, format: :json
+        end.to change { user.events.count }.by(1)
+
+        event = user.events.last
+
+        expect(event.description).to eq('Desccription for event')
+        expect(event.status).to eq('scheduled')
+        expect(event.start_date.in_time_zone).to eq(Time.zone.local(2019, 'jan', 1))
+        expect(event.finish_date.in_time_zone).to eq(Time.zone.local(2019, 'jan', 2))
+        expect(event.comments).to eq('Comment about event')
+        expect(event.summary).to eq('Summary about event')
+        expect(event.targets.count).to eq(2)
+
+        expect(response_json['status']).to eq('created')
+      end
 
-      expect { post :create, params: params, format: :json }.to change { ActionMailer::Base.deliveries.count }.by(1)
-      expect(Event.where(description: 'some description')).to exist
+      it 'send email when created event' do
+        expect do
+          post :create, params: { event: event_params }, format: :json
+        end.to change { ActionMailer::Base.deliveries.count }.by(1)
+        expect(Event.where(description: 'Desccription for event')).to exist
+      end
+
+      it 'unprocessable response when finish date less than start date' do
+        expect do
+          post(
+            :create,
+            params: { event: event_params.merge(finish_date: event_params[:start_date] - 2.days) },
+            format: :json
+          )
+        end.to change(Event, :count).by(0)
+        expect(response_json['finish_date'].first).to eq("can't be less than start date")
```

Comment:
if `response_json['finish_date']` is array use `include`:
https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/include-matcher